internal/r8/deobfuscator.go

```diff
@@ -0,0 +1,205 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package r8
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+)
+
+type StacktraceType struct {
+	name        string
+	indentation int
+	methods     map[string]string // Maps method references to their stacktrace call site
+}
+
+type MappedType struct {
+	obfuscated StacktraceType
+	realName   string
+}
+
+type MappedMethodCall struct {
+	reference string
```

Comment:
a comment to describe `reference` would be good. It is `obfuscatedTypeName:obfuscatedMethodName(:sourcefileStart)` if i understand correctly